apps/astarte_trigger_engine/lib/astarte_trigger_engine/events_consumer.ex

```diff
@@ -62,41 +62,65 @@ defmodule Astarte.TriggerEngine.EventsConsumer do
     handle_simple_event(realm, device_id, headers, event_type, event, timestamp_ms)
   end
 
-  defp handle_simple_event(realm, device_id, headers_map, event_type, event, timestamp_ms) do
+  defp handle_simple_event(
+         realm,
+         device_id,
+         headers_map,
+         event_type,
+         event,
+         timestamp_ms
+       ) do
     with {:ok, trigger_id} <- Map.fetch(headers_map, "x_astarte_parent_trigger_id"),
-         {:ok, action} <- retrieve_trigger_configuration(realm, trigger_id),
-         {:ok, payload} <-
-           event_to_payload(realm, device_id, event_type, event, action, timestamp_ms),
-         {:ok, headers} <- event_to_headers(realm, device_id, event_type, event, action),
-         :ok <- execute_action(payload, headers, action) do
-      :telemetry.execute(
-        [:astarte, :trigger_engine, :http_action_executed],
-        %{},
-        %{realm: realm, status: :ok}
-      )
+         {:ok, action} <- retrieve_trigger_configuration(realm, trigger_id) do
+      with {:ok, payload} <-
```

Comment:
Actually, why is there a `with` inside a `with`? If the two error cases needs different handling, consider extracting the internal one to a new function.
This refactoring suggestion might come in hand: https://github.com/lucasvegi/Elixir-Code-Smells?tab=readme-ov-file#complex-else-clauses-in-with